### PR TITLE
Authenticated users can change their login password

### PR DESCRIPTION
**Pull request by goodnessaig1**

### CODE REVIEW

**Comment by courage173** [comment]
@goodnessaig1 why is the last name saying too long while creating a user

File: src/Auth/Actions/userActions.js [hunk 1]
```diff
@@ -84,6 +85,34 @@ export const LoginUser = (credentials, history, setFieldError, setSubmitting) =>
     }
 }
 
+
+export const ChangeUserPassword = (credentials, history) => {
+    // Make checks and get some
```
Comment (courage173) [inline src/Auth/Actions/userActions.js]: add your api call in its own function in a seperate file, then you can pass the url and method to the function. call me if you don't understand

File: src/Auth/Actions/userActions.js [hunk 2]
```diff
@@ -84,6 +85,34 @@ export const LoginUser = (credentials, history, setFieldError, setSubmitting) =>
...
+        ).then((response) => {
+            const { data } = response;
+            if (data.status === "Failed") {
+                const { message} = data;
+                alert(message)
+            } else if (data.status === "success") {
+                history.push("/change_password_success")
+            }
+        }).catch(error =>{
+        const errorMsg = error
```
Comment (courage173) [inline src/Auth/Actions/userActions.js]: why are you not doing anything or showing the error to the user

**Review by courage173 — CHANGES_REQUESTED** [review]

**Comment by goodnessaig1** [comment]
> @goodnessaig1 why is the last name saying too long while creating a user

I have fixed it sir

File: src/Auth/Actions/userActions.js (hunk 1 above)
Comment (goodnessaig1) [reply src/Auth/Actions/userActions.js]: Ok sir

File: src/Components/Profile/ChangePassword.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        confirmPassword: ""
+    });
+    const makeRequest = (formData) => {
+        ChangeUserPassword(formData, history)
+    }
+
+    const handleNextStep = (newData, final = false) =>{
+        setData(prev => ({...prev, ...newData}))
+        if (final) {
+            makeRequest(newData)
```
Comment (courage173) [inline src/Components/Profile/ChangePassword.js]: await the makeRequest and also handle error, use this package `toastify` to notify errors

File: src/Components/Profile/ChangePasswordSuccess.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { useHistory, } from 'react-router-dom'
+import Good from "../Assets/good.png"
+
+const ChangePasswordSuccess = () => {
+    const history = useHistory()
+    useEffect(() => {
+        setTimeout(() => {
+            history.push('/profile')
+        }, 8000);
+        // eslint-disable-next-line
```
Comment (courage173) [inline src/Components/Profile/ChangePasswordSuccess.js]: why did you disable next line. remove and add missing dependencies

File: src/Auth/Actions/userActions.js [hunk 5]
```diff
@@ -76,40 +69,45 @@ export const LoginUser = (credentials, history, setFieldError, setSubmitting) =>
...
-                dispatch(loginUserSuccess(userData))
+                return dispatch => {
+                    dispatch(loginUserSuccess(data))
+                }
             }
             setSubmitting(false);
-        })
-        .catch(err => console.error(err));
+    } catch (error) {
+        throw new Error(error)
```
Comment (courage173) [inline src/Auth/Actions/userActions.js]: you should dispatch the error to store and show to the user

File: src/Auth/Actions/userActions.js [hunk 6]
```diff
@@ -76,40 +69,45 @@ export const LoginUser = (credentials, history, setFieldError, setSubmitting) =>
...
-        
-        return () => {
-        axios.post(`${SERVER}auth/v1/create-user`,
-        credentials,
-        {
-            headers: {
-                "Content-Type": "application/json"
-            }
+
+export const  getUserDetails = async () => {
```
Comment (courage173) [inline src/Auth/Actions/userActions.js]: rewrite like this. notice where the dispatch is. the way you are writing you have to then declare dispatch twice. I think it was my fault am sorry for that

`export function getUser() {
    return dispatch => {
        const promise = getApi('auth/profile');
        dispatch(getUserRequest());
        promise.then(
            function (payload) {
                dispatch(getUserSuccess(payload.data));
            },
            function (error) {
                dispatch(getUserError(error));
            }
        );

        return promise;
    };
}`

**Review by courage173 — CHANGES_REQUESTED** [review]

File: src/Auth/Actions/userActions.js (hunk 5 above)
Comment (goodnessaig1) [reply src/Auth/Actions/userActions.js]: Ok Sir

File: src/Components/Profile/ChangePasswordSuccess.js (hunk 4 above)
Comment (goodnessaig1) [reply src/Components/Profile/ChangePasswordSuccess.js]: Ok sir

File: src/Auth/Actions/userActions.js (hunk 6 above)
Comment (goodnessaig1) [reply src/Auth/Actions/userActions.js]: Ok sir

**Review by courage173 — APPROVED** [review]

**Pull request merged**